src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -597,6 +598,54 @@ def test_automation_job_add_commit_success_git(
     assert len(job.landed_commit_id) == 40, "Landed commit ID should be a 40-char SHA."
 
 
+@pytest.mark.django_db
+def test_automation_job_add_commit_base64_success_git(
+    treestatusdouble, git_automation_worker, repo_mc, monkeypatch, git_patch
+):
+    repo = repo_mc(SCM_TYPE_GIT)
+    scm = repo.scm
+
+    # Create a valid patch and base64-encode it.
+    patch_text = git_patch()
+    patch_bytes = patch_text.encode("utf-8")
+    patch_b64 = base64.b64encode(patch_bytes).decode("ascii")
+
+    # Create a job and the new base64 commit action.
+    job = AutomationJob.objects.create(
+        status=JobStatus.SUBMITTED,
+        requester_email="example@example.com",
+        target_repo=repo,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="add-commit-base64",
+        data={
+            "action": "add-commit-base64",
+            "content": patch_b64,
+        },
+        order=0,
+    )
```

Comment:
Ideally these would go into a fixture that can be customized as needed, instead of recreating jobs/actions for every test from scratch.

Reply:
Agreed, there is a lot of repeated boilerplate here as well. This would be a larger change and out of scope for this PR, filed https://bugzilla.mozilla.org/show_bug.cgi?id=1973247 to track this.